pkg/common/config.go

```diff
@@ -0,0 +1,58 @@
+package common
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/promlog"
+	promlogflag "github.com/prometheus/common/promlog/flag"
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
+	"gopkg.in/yaml.v2"
+)
+
+type ProbeConfig struct {
+	LogConfig      promlog.Config `yaml:"log,omitempty"`
+	HttpListenAddr string         `yaml:"http_listen_addr,omitempty"`
+	ConfigPath     string         `yaml:"config_path,omitempty"`
+}
+
+func AddFlags(a *kingpin.Application, cfg *ProbeConfig) {
+	a.HelpFlag.Short('h')
+	a.Flag("web.listen-address", "Address to listen on for UI, API, and telemetry.").
+		Default("0.0.0.0:8080").StringVar(&cfg.HttpListenAddr)
+	a.Flag("config.path", "Path to the probe configuration file").
+		Default("config.yaml").StringVar(&cfg.ConfigPath)
```

Comment:
IMO you should either have a config.yaml file or change the default to an existing one